tests/test_datasette_https_server.sh

```diff
@@ -13,21 +17,48 @@ datasette --memory \
 # Store the background process ID in a variable
 server_pid=$!
 
+test_url='https://localhost:8152/_memory.json'
+
 # Wait for the server to start
-sleep 2
+
+# h/t https://github.com/pouchdb/pouchdb/blob/25db22fb0ff025b8d2c698da30c6c409066baa0c/bin/run-test.sh#L102-L113
+waiting=0
+until $(curl --output /dev/null --silent --insecure --head --fail --max-time 2 $test_url); do
+    if [ $waiting -eq 4 ]; then
+        echo "$test_url can not be reached, server failed to start"
+        cleanup
+        exit 1
+    fi
+    let waiting=waiting+1
+    sleep 1
+done
 
 # Make a test request using curl
-curl -f --cacert client.pem 'https://localhost:8152/_memory.json'
+curl -f --cacert client.pem $test_url
 
 # Save curl's exit code (-f option causes it to return one on HTTP errors)
 curl_exit_code=$?
 
 # Shut down the server
 kill $server_pid
-sleep 1
+waiting=0
+#         show all pids
+#         |       find just the $server_pid
+#         |       |                  don’t match on the previous grep
+#         |       |                  |            we don’t need the output
+#         |       |                  |            |
+until ( ! ps ax | grep $server_pid | grep -v grep > /dev/null ); do
```

Comment:
This can be simplified in a few different ways:

```sh
until ! ps -p $server_pid > /dev/null; do

until ! [ -e /proc/$server_pid ]; do

until ! kill -0 $server_pid 2>/dev/null; do
```

These have varying degrees of portability - don't have a Mac to hand, but I'm guessing they all work on all modern systems.  They all assume that `$server_pid` cannot be the empty string, but I think that's safe in this case.